tests/prague/eip7692_eof_v1/eip4200_relative_jumps/test_rjumpi.py

```diff
@@ -827,3 +827,37 @@ def test_rjumpi_backwards_reference_only(
         data=container,
         expect_exception=EOFException.UNREACHABLE_INSTRUCTIONS,
     )
+
+
+def test_rjumpi_stack_validation(
+    eof_test: EOFTestFiller,
+):
+    """
+    Check that you can get to the same opcode with two different stack heights
+    Spec now allows this:
+    4.b in https://github.com/ipsilon/eof/blob/main/spec/eof.md#stack-validation
+    """
+    container = Container.Code(code=Op.RJUMPI[1](1) + Op.ADDRESS + Op.NOOP + Op.STOP)
+    eof_test(
+        data=container,
+        expect_exception=None,
+    )
+
+
+def test_rjumpi_at_the_end(
+    eof_test: EOFTestFiller,
+):
+    """
+    https://github.com/ipsilon/eof/blob/main/spec/eof.md#stack-validation 4.i:
+    This implies that the last instruction may be a terminating instruction or RJUMP
+    """
+    eof_test(
+        data=Container(
+            sections=[
+                Section.Code(
+                    code=Op.PUSH1(0) + Op.PUSH1(0) + Op.RJUMPI[1] + Op.STOP + Op.RJUMPI[-3],
```

Comment:
```suggestion
                    code=Op.PUSH1(0) + Op.PUSH1(0) + Op.RJUMPI[1] + Op.STOP + Op.RJUMPI[-4],
```
I would put this to jump to the `Op.STOP` because jumping back to itself also produces stack underflow.